### PR TITLE
Visualization Fixes #2

### PR DESCRIPTION
**Pull request by marcusnovotny**
Replaces pull request #82. Working on #71.

**Provided functionality:**
- Start on time-range with data. Prevent zooming / panning out of our data-range.
- Mark episode airing dates on the x-axis
- detect CSV path prefix (not always /csv, depends on the implementation) from original passed CSV path

Also fixes old zooming & panning bugs. 

**Known issues:**
- Overlapping airing dates / episodes on the x-axis: Normally there's at least a week between episodes, then the code works just fine. Unfortunately, some episodes are saved with the wrong airing dates [#123](https://github.com/Rostlab/JS16_ProjectA/issues/123). Waiting for Project A fix.

Please review & merge :)

### CODE REVIEW

File: public/chart.css [hunk 1]
```diff
@@ -21,3 +21,7 @@
     cursor: move;
     pointer-events: all;
 }
+.episode {
+    fill: grey;
```
Comment (julienschmidt) [inline public/chart.css:25]: Please use HEX or RGB values. `grey` might look very different depending on the browser defaults

File: public/chart.js [hunk 2]
```diff
@@ -2,9 +2,29 @@
...
+    var zoom = d3.behavior.zoom(); // Zooming behavior
+    this.fullYdomain = [];
+    this.fullXdomain = [];
+    this.prefix = ""; // CSV Path Prefix
+    this.resize = render; // Resizing behaviour
+    this.episodeData = null;
+    this.ready = 0;
+
+    // Get CSV prefix
+    (function () {
```
Comment (julienschmidt) [inline public/chart.js:15]: Why did you wrap this in a self-invoking function?

**Comment by julienschmidt** [comment]
See my annotations for the next PR

Otherwise :+1:

File: public/chart.js (hunk 2 above)
Comment (marcusnovotny) [inline public/chart.js:15]: Because I define 3 variables that I don't need for anything else (s, p, path)

Comment (julienschmidt) [inline public/chart.js:15]: It would be better coding style to declare this as a helper function somewhere and call it by name then.

Comment (julienschmidt) [inline public/chart.js:15]: e.g. `self.prefix = findPath(dataUrl);`

**Pull request merged**